### PR TITLE
Task/remove document error on invalid submission

### LINKED ISSUE #277: FormBuiler related error on Submission validation error
NoMethodError at /grants/<NAME>/submissions
undefined method `remove_document' for #<GrantSubmission::Response:XXXXXXXXXX>

This was found while submitting a submission. To duplicate, have a grant with a document upload and a integer question. Then apply but put text in the int field after uploading a document. 

- [x] GrantSubmission::Question factory to include file_upload trait
- [x] Update GrantSubmission::Question model specs to include file_upload
- [x] GrantSubmission::Response factory for valid and invalid file_upload types
- [x] GrantSubmission::Response model specs for file_upload
- [x] Add spec for successful submission with file_upload question and response 
- [x] Add spec for failed submission with file_upload question and response 
- [x] Add spec for failed submission with valid file_upload response and an invalid response to another field

### PR DESCRIPTION
**Pull request by jseraf**
closes #277 

- Adds GrantSubmission::Question response_type file_upload factories and specs
- Adds GrantSubmission::Response file_upload factories and specs
- Update GrantSubmission::Response system spec to add file_upload question/response
- Restore remove_document method from FormBuilder

### CODE REVIEW

File: spec/system/grant_submissions/grant_submission_responses_spec.rb [hunk 1]
```diff
@@ -11,69 +15,115 @@
       find_field('Your Project\'s Title', with: '').set(Faker::Lorem.sentence)
     end
 
-    scenario 'requires response mandatory short_text question' do
-      short_text_question = @grant.questions.where(response_type: 'short_text').first
-      short_text_question.update_attribute(:is_mandatory, true)
+    context 'valid submission' do
+      scenario 'accepts a pick_one response' do
+        multiple_choice_question = @grant.questions.where(response_type: 'pick_one').first
```
Comment (mattbaumann1) [inline spec/system/grant_submissions/grant_submission_responses_spec.rb]: change to pick_one and other places in the file.

File: spec/models/grant_submission_response_spec.rb [hunk 2]
```diff
@@ -9,22 +10,72 @@
...
+          expect(response.errors).to include(:grant_submission_question_id)
+        end
+
+        it 'validates length' do
+          response.update_attribute(:string_val, Faker::Lorem.characters(number: 256))
+          expect(response).not_to be_valid
+          expect(response.errors).to include(:string_val)
+        end
+
+        it 'validates if attachment' do
```
Comment (mgurley) [inline spec/models/grant_submission_response_spec.rb]: Maybe this is overkill but maybe have this covered for all invalid response_type.  Seems to be only applied for 'short_text'

File: app/models/grant_submission/response.rb [hunk 3]
```diff
@@ -185,5 +186,9 @@ def parse_raw_value_as_a_number(raw_value)
         end
       end
     end
+
+    def attachment_is_valid
+      errors.add(:document, :not_a_file_upload, question: question) unless question.response_type == 'file_upload'
```
Comment (mgurley) [inline app/models/grant_submission/response.rb]: Maybe use an Enum or constant for response_type.  Having hardcoded strings in more than one place is not ideal.

File: spec/system/grant_submissions/grant_submission_responses_spec.rb (hunk 1 above)
Comment (jseraf) [reply spec/system/grant_submissions/grant_submission_responses_spec.rb]: agreed. will move to its own ticket, though.

File: spec/models/grant_submission_response_spec.rb (hunk 2 above)
Comment (jseraf) [reply spec/models/grant_submission_response_spec.rb]: Yep. Will open this as a new ticket and add specs for other response types.

File: spec/system/grant_submissions/grant_submission_responses_spec.rb (hunk 1 above)
Comment (jseraf) [reply spec/system/grant_submissions/grant_submission_responses_spec.rb]: See #290

File: spec/models/grant_submission_response_spec.rb (hunk 2 above)
Comment (jseraf) [reply spec/models/grant_submission_response_spec.rb]: See #291

**Pull request merged**